dsd/search.py

```diff
@@ -1163,23 +1163,25 @@ def _check_design(design: dc.Design) -> Dict[Domain, Strand]:
 
     return domain_to_strand
 
-
-def search_for_dna_sequences(*, design: dc.Design,
-                             probability_of_keeping_change: Optional[Callable[[float], float]] = None,
-                             random_seed: Optional[int] = None,
-                             never_increase_weight: Optional[bool] = None,
-                             out_directory: Optional[str] = None,
-                             report_delay: float = 60.0,
-                             on_improved_design: Callable[[int], None] = lambda _: None,
-                             restart: bool = False,
-                             force_overwrite: bool = False,
-                             debug_log_file: bool = False,
-                             info_log_file: bool = False,
-                             report_only_violations: bool = True,
-                             max_iterations: Optional[int] = None,
-                             max_domains_to_change: int = 1,
-                             num_digits_update: Optional[int] = None,
-                             ) -> None:
+@dataclass
+class SearchParameters:
```

Comment:
The docstrings for the many parameters to `search_for_dna_sequences` should be cut and pasted to docstrings for each of the fields in `SearchParameters`, e.g., instead of 

```python
:param probability_of_keeping_change:
    Function giving the probability of keeping a change in one :any:`Domain`'s DNA sequence, 
    if the new sequence affects the total weight of all violated
```

Now in `SearchParameters`, you instead should have

```python
probability_of_keeping_change: Optional[Callable[[float], float]] = None
"""
Function giving the probability of keeping a change in one :any:`Domain`'s DNA sequence, 
if the new sequence affects the total weight of all violated
"""
```

Also, make sure the docs compile. This is described in the [CONTRIBUTING document](https://github.com/UC-Davis-molecular-computing/scadnano-python-package/blob/dev/CONTRIBUTING.md) for scadnano. You cd into the docs folder and run "make html". You might have to install sphinx first via "pip install sphinx sphinx_rtd_theme"to get it to work.